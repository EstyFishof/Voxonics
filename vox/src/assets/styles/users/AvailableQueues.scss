.table-container {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    align-items: flex-start;
  
    .table-title {
      font-weight: bold;
      font-size: 32px;
      line-height: 37px;
      color: var(--header-user-color);
      padding: 16px 0;
    }
  
    .wrap-table {
      width: 766px;
  
      .table-header {
        padding-left: 16px;
        height: 45px;
        width: 100%;
        color: var(--white);
        display: flex;
        align-items: center;
        font-size: 14px;
        background: var(--light-blue);
        box-shadow: 0 4px 4px rgba(0, 0, 0, 0.12);
        border-radius: 4px 4px 0 0;
      }
      .header-name{
        width: calc(30% - 4px);
      }
      .header-number{
        width: calc(30% - 4px);
        text-align: center;
      }
      .header-length{
        width: calc(20% - 4px);
        text-align: center;
      }
  
      .header-item-action {
        padding-left: 14px;
        width: calc(20% - 4px);
        text-align: left;
      }

      .list-group {
        max-height: 200px;
        overflow-y: auto;
        padding: 0 16px 0 16px;
        font-size: 16px;
        background: var(--permission-table-bg);
        
  
        .label-container {
          position: relative;
          margin-right: 10px;
  
          input[type="checkbox"] {
            opacity: 0;
            color: transparent;
            margin-right: 10px;
            z-index: 9;
            position: relative;
  
            &:checked ~ .text {
              color: var(--light-blue);
            }
  
            &:checked ~ .check {
              position: absolute;
              opacity: 1;
              top: 0;
              left: 2px;
              width: 16px;
              height: 16px;
              background-image: url("../../../assets/img/icons/checked.svg");
              background-size: cover;
            }
  
            & ~ .check {
              position: absolute;
              opacity: 1;
              top: 0;
              left: 2px;
              width: 16px;
              height: 16px;
              background-image: var(--unchecked);
              background-size: cover;
            }
          }
        }
  
      }
  
      .list-group-item {
        color: var(--font-panel);
        display: flex;
        align-items: center;
        margin-top: 1px;
        margin-bottom: 5px;
        border: none;
        height: 45px;
        position: relative;
  
        &:after {
          content: "";
          position: absolute;
          bottom: 0;
          width: 100%;
          height: 2.01px;
          background: var(--bottom-line);
          border-radius: 2px;
  
        }
  
        .action {
          width: 20%;
          text-align: right;
        }
      }
  
      .department-name {
        width: 30%;
        text-align: left;
      }
  
      .department-user-uid {
        width: 30%;
        text-align: center;
      }
  
      .department-queue-length{
        width: 20%;
        text-align: center;
      }
    }
  }
