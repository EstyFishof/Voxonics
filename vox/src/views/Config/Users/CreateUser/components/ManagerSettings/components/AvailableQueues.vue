<template>
  <section class="table-container" v-if="getRole !== 'admin' && getRole !== 'owner'">
    <div class="table-title">Available Queues</div>
    <div class="wrap-table">
      <ul class="table-header">
        <li class="header-item header-name" @click="sortFunc('name', 'queueList', 'getRelatedQueuesName')">
          Name of Queue
          <img ref="getRelatedQueuesName" src="@/assets/img/icons/sort-arrow.svg"/>
        </li>
        <li class="header-item header-number"
            @click="sortFunc('internalNumber', 'queueList', 'getRelatedQueuesNumber')">
          Queue Number
          <img ref="getRelatedQueuesNumber" src="@/assets/img/icons/sort-arrow.svg"/>
        </li>
        <li class="header-item header-length" @click="sortFunc('userCount', 'queueList', 'getRelatedQueuesAgents')">
          Agents in Queue
          <img ref="getRelatedQueuesAgents" src="@/assets/img/icons/sort-arrow.svg"/>
        </li>
        <li class="header-item-action" @click="sortFunc('edit', 'queueList', 'getRelatedQueuesAction')">
          Actions
          <img ref="getRelatedQueuesAction" src="@/assets/img/icons/sort-arrow.svg"/>
        </li>
      </ul>
      <div>
        <div class="list-group" :class="{'add-shadow': queueList.length > 4}" @scroll="funScroll($event)">
          <ul class="list-group-item" v-for="que in queueList" :key="que.uid.slice(0, 10)">
            <li class="department-name ellipsis">{{ que.name }}</li>
            <li class="department-user-uid">{{ que.internalNumber }}</li>
            <li class="department-queue-length">{{ que.relatedWith.user.length }}</li>
            <li class="action">
              <label class="label-container">
                <input @change.prevent="chooseDep(que)" class="view" :checked="que.view" type="checkbox"/>
                <span class="check"></span>
                <span class="text">View</span></label>
              <label class="label-container">
                <input @change.prevent="setEdit(que)" class="edit" :checked="que.edit" type="checkbox"/>
                <span class="check"></span>
                <span class="text">Edit</span>
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>

  </section>
</template>

<script>
// import {QueuesGettersApi} from '@/API/getters'
import { serverSort, clientSort } from '@/lib/sort.js'
import { mapGetters } from "vuex"

export default {
  name: "AvailableQueues",
  data() {
    return {
      queueList: [],
      selectedQueueList: [],
      order: {
        name: 'asc',
        internalNumber: 'asc',
        userCount: 'asc'
      },
      activeSort: 'internalNumber',
      perPage: 9,
    }
  },
  watch: {
    getGlobalQueuesList() {
      this.$store.dispatch('queuesRelated', JSON.parse(JSON.stringify(this.getGlobalQueuesList)))
    },
    getRelatedQueues() {
      if (this.getQueueManaged.length > 0) {
        this.selectedQueueList = this.getQueueManaged
      }
      this.createDataForView()
    },
    getQueueManaged() {
      this.createDataForView()
    }
  },
  computed: {
    ...mapGetters(['getRelatedQueues', 'getQueueManaged', 'getRole', 'getPaginatorQueues', 'getGlobalQueuesList'])
  },
  methods: {
    funScroll({target}) {
      if (target.clientHeight + target.scrollTop === target.scrollHeight) {
        if (this.getPaginatorQueues.paginator.count > this.perPage) {
          this.perPage += 9
          this.fetch(this.perPage, this.activeSort, this.order[this.activeSort] === 'desc' ? 'asc' : 'desc')
        }
      }
    },
    fetch(peerPage, sort, order) {
      this.$store.dispatch('setGlobalQueuesList', {peerPage, sort, order})
    },
    sortFunc(param, arrayParam, refElement) {
      if (param !== 'edit') {
        const dataSort = serverSort(this.fetch, this.$refs[refElement], this.order[param], this.perPage, param)
        this.order[param] = dataSort.order
        this.$refs[refElement] = dataSort.refArrow
        this.activeSort = param
      } else {
        const dataSort = clientSort(param, this[arrayParam], this.$refs[refElement], this.order[param])
        this[arrayParam] = dataSort.array
        this.order[param] = dataSort.order
        this.$refs[refElement] = dataSort.refArrow
      }
    },
    chooseDep(que) {
      const includes = this.selectedQueueList.filter(d => d.uid === que.uid)
      if (!includes.length) {
        que.view = true
        this.selectedQueueList.push(que)
      } else {
        this.selectedQueueList = this.selectedQueueList.filter(q => q.uid !== que.uid)
        if (que.edit) {
          que.edit = false
          this.queueList.forEach((de, i) => {
            if (que.uid === de.uid) {
              de.edit = false
              //Vue2 не видит изменений этих полей, поэтому подменяем сам объект
              const dd = {...de}
              this.queueList.splice(i, 1, dd)
            }
          })
        }
      }
      this.$store.dispatch('queueManagerSelected', this.selectedQueueList)
    },
    setEdit(que) {
      const queue = Object.assign({}, que)
      if (!queue.view) {
        queue.view = true
        queue.edit = true
        this.queueList.forEach((de, i) => {
          if (queue.uid === de.uid) {
            de.view = true
            de.edit = true
            //Vue2 не видит изменений этих полей, поэтому подменяем сам объект
            const dd = {...de}
            this.queueList.splice(i, 1, dd)
          }
        })
      } else {
        queue.edit = !queue.edit
      }
      this.selectedQueueList = this.selectedQueueList.filter(q => q.uid !== queue.uid)
      this.selectedQueueList.push(queue)
      this.$store.dispatch('queueManagerSelected', this.selectedQueueList)
    },
    createDataForView() {
      this.queueList = []
      this.getRelatedQueues.forEach(que => {
        que.view = false
        que.edit = false
        let f = false
        if (this.getQueueManaged.length > 0) {
          this.getQueueManaged.forEach(item => {
            let d = JSON.parse(JSON.stringify(item))
            if (d.uid === que.uid) {
              f = true
              d.view = true
              if (d.edit) d.view = true
              this.queueList.push(d)
            }
          })
          if (!f) {
            this.queueList.push(que)
          }
        } else {
          this.queueList.push(que)
        }
      })
    }
  },
  mounted() {
    //Загружаем выбраный массив из стора
    if (this.getQueueManaged.length > 0) {
      this.selectedQueueList = this.getQueueManaged
    }
    //если есть данные в сторе, применяем их для отображения   
    this.createDataForView()
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/users/AvailableQueues.scss"></style>
