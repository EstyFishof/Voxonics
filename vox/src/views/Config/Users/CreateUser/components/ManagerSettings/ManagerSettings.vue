<template>
  <div :class="{disable: getRole !== 'manager'}" v-if="getRole !== 'agent' && getRole != undefined && getRole != null"
       class="wrap-manager-setting">
    <div class="wrap-head">
      <h2 class="wrap-head-title">MANAGER SETTINGS</h2>
      <svg @click="showOptions = !showOptions" class="arrow" :class="{'down': showOptions}" viewBox="0 0 10 18"
           fill="transparent"
           xmlns="http://www.w3.org/2000/svg">
        <path d="M9 1L1 9L9 17" stroke-width="2" stroke="var(--light-blue)" stroke-linecap="round"
              stroke-linejoin="round"/>
      </svg>
      <span class="hr"></span>
    </div>

    <template v-if="showOptions">
      <Departments/>
      <AvailableQueues/>
      <Permission/>
    </template>
  </div>
</template>

<script>
import AvailableQueues from "@/views/Config/Users/CreateUser/components/ManagerSettings/components/AvailableQueues"
import Departments from "@/views/Config/Users/CreateUser/components/ManagerSettings/components/Departments"
import Permission from "@/views/Config/Users/CreateUser/components/ManagerSettings/components/Permission"
import { mapGetters } from "vuex"

export default {
  name: "ManagerSettings",
  components: { Departments, AvailableQueues, Permission },
  data() {
    return {
      showOptions: true
    }
  },
  computed: {
    ...mapGetters(['getRole'])
  }
}
</script>

<style scoped lang="scss">
.wrap-manager-setting {
  margin-top: 36px;
  padding-bottom: 36px;
}

.disable {
  pointer-events: none;

  .wrap-head {
    svg {
      transform: rotate(180deg);

      path {
        stroke: #2B343D;
      }
    }
  }
}
</style>
