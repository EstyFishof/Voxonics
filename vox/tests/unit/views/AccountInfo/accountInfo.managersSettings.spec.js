import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import ManagerSettings from '../../../../src/views/Config/Users/Accountinfo/components/ManagerSettings/ManagerSettings';
import AvailableQueues
    from '../../../../src/views/Config/Users/Accountinfo/components/ManagerSettings/components/AvailableQueues';
import Departments
    from '../../../../src/views/Config/Users/Accountinfo/components/ManagerSettings/components/Departments';
import QManaging from '../../../../src/views/Config/Users/Accountinfo/components/ManagerSettings/components/QManaging';
import user from "../EditUser/__mock__/user";
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);

const mockDepartment = {
    uid: "71b966fb-c930-481e-aa3a-3a942a6d3e59",
    name: "Manage",
    option: {
        edit: 1
    }
};

const mockQueue = {
    uid: "e065083b-cbf0-4ce6-a11b-dfdb551597fe",
    name: "Manage queue",
    internalNumber: "1001",
    relatedWith: { user: [] },
    option: {
        edit: 1
    }
};
const mockQueue2 = {
    uid: "e065083b-cbf0-4ce6-a11b-dfdb551597fe",
    name: "Manage queue",
    internalNumber: "1002",
    relatedWith: { user: [] },
    option: {
        edit: 0
    }
};

describe('Managers settings', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters, store;
    beforeEach(() => {
        getters = {
            getRole: () => 'agent',
        };
        store = new Vuex.Store({ getters });
    });

    it('Check snapshot', () => {
        const wrapper = shallowMount(ManagerSettings, { localVue, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('should  Vue instance', () => {
        const wrapper = shallowMount(ManagerSettings, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('Vuex data defined', () => {
        const wrapper = shallowMount(ManagerSettings, { localVue, store });
        expect(wrapper.vm.getRole).toBeDefined();
    });
    it('show options  defined', () => {
        const wrapper = shallowMount(ManagerSettings, { localVue, store });
        expect(wrapper.vm.showOptions).toBeDefined();
    });
});

describe('Managers settings ==> AvailableQueues', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters;
    let store;
    let actions;
    beforeEach(() => {
        getters = { getRelatedQueues: () => [mockQueue], getQueueManaged: () => [mockQueue] };
        actions = { queueManagerSelected: jest.fn() };
        store = new Vuex.Store({ actions, getters });
    });
    it('should  instance of Vue', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('should snapshot', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('Vuex data defined', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        expect(wrapper.vm.getRelatedQueues).toBeDefined();
        expect(wrapper.vm.getQueueManaged).toBeDefined();
    });
    it('queue list  defined', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        expect(wrapper.vm.queueList).toBeDefined();
    });
    it('selected queue list  defined', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        expect(wrapper.vm.selectedQueueList).toBeDefined();
    });
    it('choose dep if includes', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        wrapper.setData({ selectedQueueList: [{ uid: '13' }] });
        const spy = jest.spyOn(wrapper.vm, 'chooseDep');
        wrapper.vm.chooseDep({ uid: '13', edit: true });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseDep if not includes', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        wrapper.setData({ selectedQueueList: [{ uid: '12345', edit: true }] });
        const spy = jest.spyOn(wrapper.vm, 'chooseDep');
        wrapper.vm.chooseDep({ uid: '13', edit: false });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setEdit  view === true', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'setEdit');
        wrapper.vm.setEdit({ uid: '13', view: true });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setEdit view === false', () => {
        const wrapper = shallowMount(AvailableQueues, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'setEdit');
        wrapper.setData({ queueList: [mockQueue] });
        wrapper.vm.setEdit({ uid: '13', view: false });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

describe('Managers settings ==> Departments / edit', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters, actions, store;
    beforeEach(() => {
        getters = {
            getRelatedDepartments: () => [mockDepartment],
            userManageDeps: () => [mockDepartment],
            getDepManaged: () => [mockDepartment],
            userInfoVuex: () => user,
            departments: () => [mockDepartment],
        };
        store = new Vuex.Store({ getters, actions: { setUserDepsManaged: jest.fn(), changeDeps: jest.fn() } });
    });
    it('it should  Vue instance', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('check snapshot dep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('vuex data defined in dep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        expect(wrapper.vm.getRelatedDepartments).toBeDefined();
        expect(wrapper.vm.userManageDeps).toBeDefined();
    });
    it('dep list  defined', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        expect(wrapper.vm.depList).toBeDefined();
    });
    it('selected deps defined', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        expect(wrapper.vm.selectedDepList).toBeDefined();
    });
    it('choose dep if includes', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        wrapper.setData({ selectedDepList: [{ uid: '123' }] });
        const spy = jest.spyOn(wrapper.vm, 'chooseManDep');
        wrapper.vm.chooseManDep({ uid: '123', edit: true });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('choose dep if not includes', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        wrapper.setData({ selectedDepList: [{ uid: '12345', edit: true }] });
        const spy = jest.spyOn(wrapper.vm, 'chooseManDep');
        wrapper.vm.chooseManDep({ uid: '123', edit: false });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setEdit / view == true', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'setEdit');
        wrapper.vm.setEdit({ uid: '123', view: true });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setEdit/ view == false', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'setEdit');
        wrapper.setData({ depList: [mockDepartment] });
        wrapper.vm.setEdit({ uid: '123', view: false });
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('editMode related', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'editMode');
        wrapper.vm.editMode('related');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('editMode', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'editMode');
        wrapper.vm.editMode('123');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('addRelDep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        wrapper.setData({ selectedRelDep: mockDepartment });
        const spy = jest.spyOn(wrapper.vm, 'addRelDep');
        wrapper.vm.addRelDep();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseRelDep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'chooseRelDep');
        wrapper.vm.chooseRelDep(mockDepartment);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteRelDep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'deleteRelDep');
        wrapper.vm.deleteRelDep(mockDepartment);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('addManDep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        wrapper.setData({ selectedManDep: mockDepartment });
        const spy = jest.spyOn(wrapper.vm, 'addManDep');
        wrapper.vm.addManDep();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

    it('deleteManDep', () => {
        const wrapper = shallowMount(Departments, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'deleteManDep');
        wrapper.vm.deleteManDep(mockDepartment);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

describe('Managers settings ==> QManaging', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    let getters;
    let store;
    let actions;
    beforeEach(() => {
        getters = {
            userManageQueues: () => [mockQueue, mockQueue2],
            getRelatedQueues: () => [mockQueue],
            queues: () => [],
        };
        actions = {
            setUserQueuesManaged: jest.fn()
        };
        store = new Vuex.Store({ actions, getters });
    });
    it('should  instance of Vue', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        expect(wrapper.vm).toBeTruthy();
    });
    it('should snapshot', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        expect(wrapper.element).toMatchSnapshot();
    });
    it('Vuex data defined', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        expect(wrapper.vm.userManageQueues).toBeDefined();
        expect(wrapper.vm.getRelatedQueues).toBeDefined();
        expect(wrapper.vm.queues).toBeDefined();
    });
    it('queue list  defined', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        expect(wrapper.vm.sort).toBeDefined();
        expect(wrapper.vm.change).toBeDefined();
        expect(wrapper.vm.showRelQueues).toBeDefined();
        expect(wrapper.vm.relQueuesValueSearch).toBeDefined();
        expect(wrapper.vm.selectedRelQueue).toBeDefined();
    });

    it('sortNumber', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'sortNumber');
        wrapper.vm.sortNumber();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sortNumber sort', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        wrapper.setData({sort: {number: true}});
        const spy = jest.spyOn(wrapper.vm, 'sortNumber');
        wrapper.vm.sortNumber('number');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sortName', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'sortName');
        wrapper.vm.sortName();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sortActions', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'sortActions');
        wrapper.vm.sortActions();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setChange', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'setChange');
        wrapper.vm.setChange();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseRelQueues', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'chooseRelQueues');
        wrapper.vm.chooseRelQueues(mockQueue);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteQueues', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        const spy = jest.spyOn(wrapper.vm, 'deleteQueues');
        wrapper.vm.deleteQueues(mockQueue);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('addQueues', () => {
        const wrapper = shallowMount(QManaging, { localVue, store });
        wrapper.setData({ selectedRelQueue: mockQueue });
        const spy = jest.spyOn(wrapper.vm, 'addQueues');
        wrapper.vm.addQueues();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});
