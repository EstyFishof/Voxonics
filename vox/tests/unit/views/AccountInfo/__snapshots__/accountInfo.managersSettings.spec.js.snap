// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Managers settings ==> AvailableQueues should snapshot 1`] = `
<section
  class="table-container"
>
  <div
    class="table-title"
  >
    Available Queues
  </div>
   
  <div
    class="wrap-table"
  >
    <ul
      class="table-header"
    >
      <li
        class="header-item"
      >
        Name of Department
      </li>
       
      <li
        class="header-item-action"
      >
        Actions
      </li>
    </ul>
     
    <div>
      <div
        class="list-group"
      />
    </div>
  </div>
</section>
`;

exports[`Managers settings ==> Departments / edit check snapshot dep 1`] = `
<section
  class="table-container"
>
  <div
    class="managed"
  >
    <div
      class="table-title"
    >
      <span>
        In Departments
      </span>
       
      <button
        class="btn-edit"
        content="Edit mode"
      >
        <img
          alt=""
          src="@/assets/img/icons/edit.svg"
        />
      </button>
    </div>
     
    <div
      class="wrap-table"
    >
      <div
        class="table-title subtitle"
        style="display: none;"
      >
        <button
          class="btn-edit btn-add"
          content="Show/hide list"
        >
          <img
            alt=""
            src="@/assets/img/icons/list.svg"
          />
        </button>
         
        <div>
          <input
            placeholder="Search Departments"
            type="text"
          />
           
          <ul
            class="listRelDeps"
            style="display: none;"
          />
        </div>
      </div>
       
      <div
        class="br"
      >
        <div
          class="list-group"
        >
          <ul
            class="list-group-item"
          >
            <li
              class="department-name"
            >
              <button
                class="btn-del"
                style="display: none;"
              >
                -
              </button>
              
              Manage
            
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="managed"
  >
    <div
      class="table-title"
    >
      <span>
        Managing Departments
      </span>
       
      <button
        class="btn-edit"
        content="Edit mode"
      >
        <img
          alt=""
          src="@/assets/img/icons/edit.svg"
        />
      </button>
    </div>
     
    <div
      class="wrap-table"
    >
      <div
        class="table-title subtitle"
        style="display: none;"
      >
        <button
          class="btn-edit btn-add"
          content="Show/hide list"
        >
          <img
            alt=""
            src="@/assets/img/icons/list.svg"
          />
        </button>
         
        <div>
          <input
            placeholder="Search Departments"
            type="text"
          />
           
          <ul
            class="listRelDeps"
            style="display: none;"
          />
        </div>
      </div>
       
      <div
        class="br"
      >
        <div
          class="list-group"
        >
          <ul
            class="list-group-item"
          >
            <li
              class="department-name"
            >
              <button
                class="btn-del"
                style="display: none;"
              >
                -
              </button>
              
              Manage
              
              <img
                class="setting-img"
                src="@/assets/img/icons/setting.svg"
              />
            </li>
             
            <li
              class="action"
            >
              <label
                class="label-container"
              >
                <input
                  checked="checked"
                  class="view"
                  disabled="disabled"
                  type="checkbox"
                />
                 
                <span
                  class="check"
                />
                 
                <span
                  class="text"
                >
                  View
                </span>
              </label>
               
              <label
                class="label-container"
              >
                <input
                  class="edit"
                  type="checkbox"
                />
                 
                <span
                  class="check"
                />
                 
                <span
                  class="text"
                >
                  Edit
                </span>
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Managers settings ==> QManaging should snapshot 1`] = `
<div
  class="queues-managing"
>
  <div
    class="queues-managing__title"
  >
    
    Available Queues
  
  </div>
   
  <div
    class="table-content"
  >
    <div
      class="table-content__sticky"
    >
      <ul
        class="table-content__list-header"
      >
        <li
          class="table-content__header-item"
        >
          <button
            class="edit-close-btn"
          >
            <img
              alt=""
              content="Edit mode"
              height="10.67"
              src="@/assets/img/icons/edit.svg"
              width="10.67"
            />
             
            <!---->
          </button>
        </li>
         
        <li
          class="table-content__header-item"
        >
          Name of Queue
        </li>
         
        <li
          class="table-content__header-item"
        >
          Queue Number
        </li>
         
        <li
          class="table-content__header-item"
        >
          Actions
        </li>
      </ul>
       
      <!---->
    </div>
     
    <ul
      class="table-content__list-main"
    >
      <li
        class="table-content__list-item delete"
        content="Remove item"
        style="display: none;"
      >
        <span
          class="delete-span"
        />
      </li>
       
      <li
        class="table-content__list-item"
      >
        Manage queue
      </li>
       
      <li
        class="table-content__list-item"
      >
        1001
      </li>
       
      <li
        class="table-content__list-item"
      >
        <label
          class="table-content__label"
        >
          <input
            type="checkbox"
          />
           
          <span
            class="check"
          />
           
          <span
            class="text"
          >
            Edit
          </span>
        </label>
         
        <span
          class="table-content__img-settings"
        >
          <img
            height="20"
            src="@/assets/img/icons/setting.png"
            width="20"
          />
        </span>
      </li>
    </ul>
    <ul
      class="table-content__list-main"
    >
      <li
        class="table-content__list-item delete"
        content="Remove item"
        style="display: none;"
      >
        <span
          class="delete-span"
        />
      </li>
       
      <li
        class="table-content__list-item"
      >
        Manage queue
      </li>
       
      <li
        class="table-content__list-item"
      >
        1002
      </li>
       
      <li
        class="table-content__list-item"
      >
        <label
          class="table-content__label"
        >
          <input
            type="checkbox"
          />
           
          <span
            class="check"
          />
           
          <span
            class="text"
          >
            Edit
          </span>
        </label>
         
        <span
          class="table-content__img-settings"
        >
          <img
            height="20"
            src="@/assets/img/icons/setting.png"
            width="20"
          />
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Managers settings Check snapshot 1`] = `<!---->`;
